rqt_py_common/src/rqt_py_common/message_tree_model.py

```diff
@@ -70,13 +82,14 @@ def _recursive_create_items(self, parent, slot, slot_name, slot_type_name, slot_
             row.append(item)
 
         is_leaf_node = False
-        # TODO(mlautman): Work around missing _slot_types in new msg types
-        if hasattr(slot, '__slots__') and hasattr(slot, '_slot_types'):
-            for child_slot_name, child_slot_type in zip(slot.__slots__, slot._slot_types):
-                child_slot_path = slot_path + '/' + child_slot_name
-                child_slot = getattr(slot, child_slot_name)
+        if hasattr(slot, '__slots__'):
+            for child_slot_name in slot.__slots__:
+                child_property_name = self._get_property_name(child_slot_name, slot)
+                child_slot_type = type(getattr(slot, child_property_name)).__name__
```

Comment:
This returns list when what we actually want is `rqt_py_common/Val[5]` I already fixed this and implemented the solution in a pr to rosidl_python